main.cpp

```diff
@@ -188,214 +286,313 @@ struct Range
 {
-    // 1) diameter of each wheel (float)
     float wheelDiameter = 3.21f;
-    // 2) number of wheels (int)
     int numWheels = 8;
-    // 3) weight of the nose landing gear (float)
     float weightOfNose = 34.2f;
-    // 4) number of the four main landing gear mechanisms (int)
     int numMainLandingGear  = 16;
-    // 5) pressure of the tires (float)
     float pressureTire = 200.1f;
 
-    // 1) dissipates the kinetic energy of landing impact
-    void reduceLandingImpact (float tirePressure = 250.2f);
-    // 2) prevents the fuselage from hitting the ground
-    void preventFuselageHittingGround (bool fuselageHitGround = false);
-    // 3) support the plane during takeoff
-    void supportDuringTakeOff (bool takeOff = true);
+    int reduceLandingImpact (float tirePressure = 250.2f, int landingSpeed = 20);
+    bool preventFuselageHittingGround (bool retractLandingGear = false);
+    bool toggleLandingGear (bool takeOff = true);
 };
 
-/*
-Thing 7) tail
- */
+int LandingGear::reduceLandingImpact (float tirePressure, int landingSpeed)
+{
+    int calculateImpact = 3 * landingSpeed;
+    if(tirePressure > 200) return calculateImpact;  
+    return landingSpeed;
+}
+bool LandingGear::preventFuselageHittingGround (bool retractLandingGear)
+{
+    if(!retractLandingGear)
+         return true;
+    return false;
+}
+bool LandingGear::toggleLandingGear (bool takeOff)
+{
+    bool retractLandingGear = true;
+    if(takeOff) 
+        return !retractLandingGear;
+    return retractLandingGear; 
+}
+
 struct PlaneTail 
 {
-    // 1) weight of lower rudder (float)
     float weightLowerRudder =  2400.1f;
-    // 2) length of lower rudder (float )
     float lengthOfLowerRudder = 421.5f;
-    // 3) things its auxiliary power unit can do (std::string)
     std::string auxPower = "helps plane turn";
-    // 4) weight of upper rudder (float)
     float weigtUpperRudder = 2400.1f;
-    // 5) height of upper rudder (float)
     float heightUpperRudder = 421.5f;
-
-    // 1) its small engine powers the jet's electrical when on the ground
-    void runSmallEngine (bool electricalsOn = true);
-    // 2) its power unit powers the air-conditioning systems when on the ground
-    void runPowerUnit (bool airConditioningOn = true);
-    // 3) its rudder section turns the airplane from right to left in flight
+    int powerConsumed = 1;
+    
+    void showAlertMsg (bool);
+    void consumePower (bool);
     void turnPlane (bool lowerRudderOpen = true);
 };
-/*
-Thing 8) passenger cabin
- */
+
+void alert()
+{
+    std::cout << "Electricals are running" << std::endl;
+}
+
+void PlaneTail::showAlertMsg (bool switchOn)
+{ 
+    // when the switch is on, alert to pilot electicals are running
+    if(switchOn)
+        alert();
+    std::cout << "Electricals are not running" << std::endl;
+}
+
+void PlaneTail::consumePower (bool engineRunning)
+{
+    while (engineRunning) {
+        std::cout << "Engine running" << "\n";
+        powerConsumed++;
+    }
+    
+}
```

Comment:
loops have not been covered yet and neither have the `++ --` operators.
You have also not provided a means of exiting the loop.  
if `engineRunning` is true, this loop will run forever.

- [ ] revise